hal/targets/hal/TARGET_ONSEMI/TARGET_NCS36510/serial_api.c

```diff
@@ -0,0 +1,406 @@
+		irq_mask |= active_irq & UART_LSR_TX_EMPTY_MASK;
+	}
+
+	if(UART2REG->IER.WORD & UART_IER_RX_DATA_READY_MASK){  /*check if RX interrupt is enabled*/
+		irq_mask |= active_irq & UART_LSR_RX_DATA_READY_MASK;
+	}
+
+	//uart_irq((uint8_t)(UART2REG->LSR.WORD & 0xFF), 1);
+	uart_irq(active_irq & irq_mask, 1);
+
+}
+
+static inline void uart_irq(uint8_t status, uint32_t index)
+{
+    if (serial_irq_ids[index] != 0)
+	{
+        if (status & UART_LSR_TX_EMPTY_MASK)
+		{
+            irq_handler(serial_irq_ids[index], TxIrq);
+		}
+        if (status & UART_LSR_RX_DATA_READY_MASK)
+		{
+			irq_handler(serial_irq_ids[index], RxIrq);
+		}
+    }
+}
+/******************************************************/
+
+void serial_irq_set(serial_t *obj, SerialIrq irq, uint32_t enable)
+{
+	IRQn_Type irq_n = (IRQn_Type)0;
+	uint32_t Vector = 0;
+
+	/* Check UART number & assign irq handler */
+	if(obj->UARTREG == UART1REG)
+	{/* UART 2 */
+		Vector = (uint32_t)&Uart1_Irq;
+		irq_n = Uart1_IRQn;
+	}
+	else if(obj->UARTREG == UART2REG)
+	{/* UART 1 */
+		Vector = (uint32_t)&Uart2_Irq;
+		irq_n = Uart2_IRQn;
+	}
+	else
+	{
+		MBED_ASSERT(False);
+	}
+
+	/* Check IRQ type & enable/disable accordingly */
+	if(enable)
+	{/* Enable */
+		if(irq == RxIrq)
+		{/* Rx IRQ */
+			obj->UARTREG->FCR.BITS.RX_FIFO_TRIG = 0x0;
+			obj->UARTREG->IER.BITS.RX_DATA_INT = True;
+		}
+		else if(irq == TxIrq)
+		{/* Tx IRQ */
+			obj->UARTREG->IER.BITS.TX_HOLD_INT = True;
+		}
+		else
+		{
+			MBED_ASSERT(False);
+		}
+		NVIC_SetVector(irq_n, Vector);
+		NVIC_EnableIRQ(irq_n);
+	}
+	else
+	{/* Disable */
+		NVIC_DisableIRQ(irq_n);
+		if(irq == RxIrq)
+		{/* Rx IRQ */
+			obj->UARTREG->IER.BITS.RX_DATA_INT = False;
+		}
+		else if(irq == TxIrq)
+		{/* Tx IRQ */
+
+			obj->UARTREG->IER.BITS.TX_HOLD_INT = False;
+		}
+		else
+		{
+			MBED_ASSERT(False);
+		}
+	}
+}
+
+int serial_getc(serial_t *obj)
+{
+	uint8_t c;
+
+	while(!obj->UARTREG->LSR.BITS.READY); 	/* Wait for received data is ready */
+	c = obj->UARTREG->RBR & 0xFF; 			/* Get received character */
+	return c;
+}
+
+void serial_putc(serial_t *obj, int c)
+{
+
+		while(!obj->UARTREG->LSR.BITS.TX_HOLD_EMPTY);/* Wait till THR is empty */
+		obj->UARTREG->THR = c; /* Transmit byte */
+
+}
+
+int serial_readable(serial_t *obj)
+{
+	return obj->UARTREG->LSR.BITS.READY;
+}
+
+int serial_writable(serial_t *obj)
+{
+	return obj->UARTREG->LSR.BITS.TX_HOLD_EMPTY;
+}
+
+void serial_clear(serial_t *obj)
+{
+	/* Reset TX & RX FIFO */
+	obj->UARTREG->FCR.WORD |= ((True << UART_FCS_TX_FIFO_RST_BIT_POS) |
+							  (True << UART_FCS_RX_FIFO_RST_BIT_POS));
+}
+
+void serial_break_set(serial_t *obj)
+{
+	obj->UARTREG->LCR.BITS.BREAK = True;
+}
+
+void serial_break_clear(serial_t *obj)
+{
+	obj->UARTREG->LCR.BITS.BREAK = False;
+}
+
+void serial_pinout_tx(PinName tx)
+{
+	/* COnfigure PinNo to drive strength of 1, Push pull and pull none */
+	fPadIOCtrl(tx, 1, 0, 1);
+}
+
+#ifdef TOOLCHAIN_ARM
```

Comment:
why function like serial_set_flow_control is protected by this macro? Is serial flow enabled for  this platform?